server/master/src/main/java/org/apache/accumulo/master/upgrade/Upgrader9to10.java

```diff
@@ -625,4 +626,39 @@ private static Path resolveRelativePath(String metadataEntry, Key key) {
       return new Path(prefix + tableId.canonical() + metadataEntry);
     }
   }
+
+  /**
+   * Resolve old relative delete markers of the form /tableId/tabletDir/[file] to
+   * UpgradeVolume/tables/tableId/tabletDir/[file]
+   */
+  static Path resolveRelativeDelete(VolumeManager fs, String oldDelete, String upgradeProperty) {
+    Path pathNoVolume = VolumeManager.FileType.TABLE.removeVolume(new Path(oldDelete));
+
+    // abs path won't be null so return
+    if (pathNoVolume != null)
+      return pathNoVolume;
```

Comment:
This says "absolute path won't be null", but that's not the same as "relative path must be null". Can relative path be non-null? If so, then this will return for both absolute paths and non-null relative paths.

Reply:
The ```removeVolume``` method will return null if it doesn't have a volume aka it is a relative path.